File: Principles_Of_Programming_Languages/pascal/word_counter.pas
program wordCount;
(*count the number of entered words by user*)

var
  words: string;
  count, i: integer;

begin
  writeln('Enter a statement: ');
  readln(words);

  count := 1;

  for i := 1 to length(words) do
    if words[i] = ' ' then
      count := count + 1;

  writeln('The number of words are: ', count);
end.

(*
Output example:

Enter a statement:
Welcome to Pascal
The number of words are: 3
*)

File: Principles_Of_Programming_Languages/prolog/word_counter.pl
%Word_Count_Program

word_count(String, Count) :- 
    split_string(String, " ", "", Words), 
    length(Words, Count). 

% Output example:
% ?- word_count("Hello world from Prolog", Count), 
%    write("Number of words = "),
%    write(Count), nl, nl.
% 
% Number of words = 4